compiler/rustc_errors/src/lib.rs

```diff
@@ -801,8 +804,6 @@ impl DiagCtxt {
     pub fn print_error_count(&self, registry: &Registry) {
         let mut inner = self.inner.borrow_mut();
 
-        inner.emit_stashed_diagnostics();
```

Comment:
Either add an assertion here, too, or leave a comment as to why there may be stashed, unemitted diagnostics.